docs/admin/auth-server/oauth-features/pkce.md

```diff
@@ -7,14 +7,232 @@ tags:
+In the case of `SPA` you cannot securely store a secret key, because its source is available in the browser. The same
+will happen with **Mobile Apps**, since decompiling the application will reveal the secret keys.
 
-## Have questions in the meantime?
+This is where `PKCE` comes in to solve this problem.
 
-While this documentation is in progress, you can ask questions through [GitHub Discussions](https://github.com/JanssenProject/jans/discussions) or the [community chat on Gitter](https://gitter.im/JanssenProject/Lobby). Any questions you have will help determine what information our documentation should cover.
+`PKCE` is an extension to the Authorization on Code flow to prevent `CSRF` and authorization code injection attacks.
 
-## Want to contribute?
+This flow is considered best practice when using **Single Page Apps (SPA)** or **Mobile Apps**.
 
-If you have content you'd like to contribute to this page in the meantime, you can get started with our [Contribution guide](https://docs.jans.io/head/CONTRIBUTING/).
\ No newline at end of file
+Let's see in the following diagram how it works:
+
+![pkce1.png](../../../assets/pkce1.png)
+
+[Source](https://sequencediagram.org/index.html#initialData=C4S2BsFMAIEEFdgAsD2AnEAvAhqFA7aAYRQBMYAxcFAd2gGppoAFAaSIFEAoLgB2zSgAxiH75g0AKoBnSGj4Dho7OLi9eCwSBFiJAKRXSEyHtkQF4AWwBGcnjLkBaAHyx1ALmLhtAa2jUAcxBCb3wfLjdeF0jPAHFIfDlcGAAqFJJyaAA1ORAAMxA5NOgVUmg0jJgiJGxwKHwAyDSI9RcDfCNEJE9jVAwcPEJK6AAlSABHeEhpCUZh6tr6xuhgFHKUgHozZHQsJpSuds7kRxcHNE8x0hA0SCEJVf8UIPwtrt2BkAJoXjQUS14wC45zahl6PS6CWEyRK+DKJA6UMOYK6p1cHjg736uC+QzIkBaUWcR3BmJ22MGxHxDCpmRyGAKchWazSG1WPgSzRJqOJKOQniytRApBhw3p+UKaFh8JqdQSjWRHV6aJi0AAkgARaAAFRQHMIpTgQiE02kOr1CS4QA)
+
+## PKCE Configuration
+
+For this flow to work, we must make the following configuration in `Jans Auth`.
+
+We need to set the `clientAuthenticationFiltersEnabled` field to `false`, it defaults to `true`.
+We can use `TUI` to make this configuration:
+
+![pkce3.png](../../../assets/pkce3.png)
+
+## Authorization Code Flow + PKCE
+
+In this flow, we will execute the following steps:
+
+1. Generate **Code Verifier** and **Code Challenge**
+2. Register new client
+3. Call `authorize` endpoint
+4. Call `token` endpoint
+
+### 1. Generate Code Verifier and Code Challenge
+
+It is necessary to generate these codes for this flow, since this data will be sent in the `authorize` and `token`
+endpoints.
+
+#### Code Verifier
+
+It is a random string of length `N`.
+
+The following command generates a `code verifier` of length `128`.
+
+```bash
+tr -dc 'A-Za-z0-9-._' </dev/urandom | head -c 128; echo
+```
+
+Output example:
+
+```bash
+root@jans:~# tr -dc 'A-Za-z0-9-._' </dev/urandom | head -c 128; echo
+MW54l5Ma7i8n6QsetI_dnJhfMQ6gIfBNJiWFINu6rtD0zvZUXG3STWoolT5HFatjWu2Vj1L-Au4PtRzkear088FJLzq.6tAg10wikJrIqn75HCJ7V1b_p8io_ugkPkkr
+```
+
+#### Code Challenge
+
+`Jans Auth` supports 2 methods (`codeChallengeMethod`):
```

Comment:
what is `codeChallengeMethod`? Is it a property? or a request param? I couldn't find reference to this anywhere else in the document.

Reply:
`codeChallengeMethod` is the method that will be used to encode or not the `codeChallenge`.
It can be `plain` or `s256`.
But I will modify that part, so that it is better understood.